gensim/models/word2vec.py

```diff
@@ -562,14 +573,6 @@ def scale_vocab(self, min_count=None, sample=None, dry_run=False):
         # print extra memory estimates
         report_values['memory'] = self.estimate_memory(vocab_size=len(retain_words))
 
-        if not dry_run:
-            new_vocab = {}
-            for w in retain_words:
-                new_vocab[w] = self.vocab[w]
-                new_vocab[w].index = len(self.index2word)
-                self.index2word.append(w)
-            self.vocab = new_vocab
-            logger.info("vocabulary min_count & sample applied, and indexes assigned")
         return report_values
```

Comment:
I don't see `raw_vocab` being dropped anywhere, which is probably the intent. A separate `raw_vocab` is definitely a win for memory (by deferring Vocab creation) and clarity, but in this approach I might miss the opportunity to apply `scale_vocab()` (with effect) more than once. (Of course, only tightening min_count works but sample could vary either way.) Perhaps non-`dry_run` invoking w/ Vocab-filled vocab already in place should reconstitute a raw_vocab, to allow another effective edit?